### DISCUSSION TITLE
Remark stringify TS error type

### DISCUSSION THREAD

**Discussion by julienfollow** [remarkjs/remark#983] [Q&A] [answered]
Hi
I've an issue with remark strignify when i tried to pass it in unified.
I read your doc and do same thing.
Version
`    "unified": "^10.1.2"`
`    "remark-stringify": "^10.0.2"`

I got this with "remark-slate-transformer" but an update version solve issue.
![image](https://user-images.githubusercontent.com/70938205/163707133-dcbd88aa-2dc6-4ab8-beb0-8d4b6a83aafd.png)
![image](https://user-images.githubusercontent.com/70938205/163707147-c53e59d2-e503-4977-ace9-eeced41edd7a.png)

Thanks a lot :+1: 

Ps:  i write error for people who search same

>   Overload 1 of 3, '(plugin: Plugin<[], Root, Node>, ...settings: [] | [boolean]): Processor<Node, Node, Node, void>', gave the following error.
    Argument of type 'Plugin<[(Pick<Options, "bullet" | "join" | "strong" | "bulletOther" | "bulletOrdered" | "bulletOrderedOther" | "closeAtx" | "emphasis" | "fence" | "fences" | "incrementListMarker" | ... 9 more ... | "unsafe"> | undefined)?] | void[], Root, string>' is not assignable to parameter of type 'Plugin<[], Root, Node>'.
      The 'this' types of each signature are incompatible.
        Type 'Processor<void, Root, void, void>' is not assignable to type 'Processor<void, Root, Root, string>'.
          Types of property 'use' are incompatible.
            Type '{ <PluginParameters extends any[] = any[], Input = import("/home/julien/Follow/fw_front_react/node_modules/@types/mdast/index").Root, Output = Input>(plugin: import("/home/julien/Follow/fw_front_react/packages/app/node_modules/unified/index").Plugin<PluginParameters, Input, Output>, ...settings: [...] | PluginParame...' is not assignable to type '{ <PluginParameters extends any[] = any[], Input = import("/home/julien/Follow/fw_front_react/node_modules/@types/mdast/index").Root, Output = Input>(plugin: import("/home/julien/Follow/fw_front_react/node_modules/unified/index").Plugin<PluginParameters, Input, Output>, ...settings: [...] | PluginParameters): import...'.
              Types of parameters 'plugin' and 'presetOrList' are incompatible.
                Type 'Preset | PluggableList' is not assignable to type 'Plugin<any, any, any>'.
                  Type 'Preset' is not assignable to type 'Plugin<any, any, any>'.
                    Type 'Preset' provides no match for the signature '(this: Processor<void, void, void, void> | Processor<void, any, void, void> | Processor<void, any, any, any> | Processor<any, any, void, void>, ...settings: any): void | Transformer<...>'.
  Overload 2 of 3, '(tuple: [Plugin<any[], Node, Node>, ...any[]] | [Plugin<any[], Node, Node>, boolean]): Processor<Node, Node, Node, void>', gave the following error.
    Argument of type 'Plugin<[(Pick<Options, "bullet" | "join" | "strong" | "bulletOther" | "bulletOrdered" | "bulletOrderedOther" | "closeAtx" | "emphasis" | "fence" | "fences" | "incrementListMarker" | ... 9 more ... | "unsafe"> | undefined)?] | void[], Root, string>' is not assignable to parameter of type '[Plugin<any[], Node, Node>, ...any[]] | [Plugin<any[], Node, Node>, boolean]'.
      Type 'Plugin<[(Pick<Options, "bullet" | "join" | "strong" | "bulletOther" | "bulletOrdered" | "bulletOrderedOther" | "closeAtx" | "emphasis" | "fence" | "fences" | "incrementListMarker" | ... 9 more ... | "unsafe"> | undefined)?] | void[], Root, string>' is not assignable to type '[Plugin<any[], Node, Node>, boolean]'.
  Overload 3 of 3, '(presetOrList: Preset | PluggableList): Processor<Node, Node, Node, void>', gave the following error.
    Argument of type 'Plugin<[(Pick<Options, "bullet" | "join" | "strong" | "bulletOther" | "bulletOrdered" | "bulletOrderedOther" | "closeAtx" | "emphasis" | "fence" | "fences" | "incrementListMarker" | ... 9 more ... | "unsafe"> | undefined)?] | void[], Root, string>' is not assignable to parameter of type 'Preset | PluggableList'.
      Type 'Plugin<[(Pick<Options, "bullet" | "join" | "strong" | "bulletOther" | "bulletOrdered" | "bulletOrderedOther" | "closeAtx" | "emphasis" | "fence" | "fences" | "incrementListMarker" | ... 9 more ... | "unsafe"> | undefined)?] | void[], Root, string>' is missing the following properties from type 'Pluggable<any[]>[]': pop, push, concat, join, and 27 more.

**Comment by julienfollow** [comment]
Ok, if i use 
import remarkStringify from 'remark-stringify/lib'
I don't have error
It's good ?

**Comment by wooorm** [comment, marked as answer]
Hi. The problem is due to `remark-slate-transformer` not defining what it accepts as input and results in as output.
It should define the input it accepts [here](https://github.com/inokawa/remark-slate-transformer/blob/e980e03b46109c47378627fbef82679527c8102c/src/plugins/slate0.47-to-remark.ts#L5) according to the types that unified plugins [here](https://github.com/unifiedjs/unified/blob/b9358691047ad72e6390dfc3c3eedc15509e01fa/index.d.ts#L488-L530). Although perhaps this is not 100% possible, as it seems to accept a slate object, which might not be a Node?
